### DISCUSSION TITLE
Is it possible to contribute to this project using Python code converted to C?

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#2876] [Q&A]
I'm really impressed with Geany since I've adopted it as my default ide (for some codes) a couple of weeks ago and I would like to contribute to it. The problem is that I don't know C enough to build any decent code.

I found some possible solutions on the internet but I'm not able to say if they're good solutions or even if they are acceptable for this project. Actually, Java could be an option, too.

So, is it possible to contribute to this project using Python code converted to C?

Thanks.

**Comment by elextr** [comment]
I'm not sure how you proposed to convert Python to C.  There are tools such as Cython, but they still require the Python runtime library to be present, and it would not likely be acceptable to Geany to always require the Python library to be present in a C project.  And the generated code is unlikely to fit into existing code in C.

If you are proposing to manually convert Python to C its not likely to save you much effort, and in fact is likely to lead to memory management problems because Python handles memory management, but C doesn't and you would be adding lots of code that is not needed in the Python (its not code that would not have been needed in any C code anyway, but what I mean is that there is nothing in the Python that prompts you to add memory management when you convert it to C, so its likely to be missed).

If you can't contribute C code, and I acknowledge its a learning curve that might not be worth it to you, have you considered other contributions, possibly cleaning up the manual, or adding better explanations of some of the features for example.  

It used to be possible to write plugins in Python, but that capability depended on features that were not continued from Python 2 to Python 3 so now its only available using an out-of-tree capability that supports Python 3.
